src/utils/CryptoUtils.ts

```diff
@@ -29,4 +37,17 @@ export class CryptoUtils {
         const decipher = crypto.createDecipheriv(this.algorithm, key, Buffer.from(iv, "hex"));
         return decipher.update(encrypted, 'hex', 'utf8') + decipher.final('utf8');
     }
+
+    decodeAuthToken(authToken: string): any {
+        let obj = {};
+        let base64Url = authToken.split('.')[1];
+        let base64  =  base64Url.replace(/-/g,'+').replace(/_/, '/')
+        let jsonPayload = decodeURIComponent(atob(base64).split('').map(function(c){
+                return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2)
+        }).join(''));
+        let jsonParse = JSON.parse(jsonPayload);
+        obj["payload"] = jsonParse;
+        return obj; 
+    }
+
```

Comment:
```suggestion
        let obj = {};
        let base64Url = authToken.split('.')[1];
        let base64  =  base64Url.replace(/-/g,'+').replace(/_/, '/')
        let jsonPayload = decodeURIComponent(atob(base64).split('').map(function(c){
                return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2)
        }).join(''));
        let jsonParse = JSON.parse(jsonPayload);
        obj["payload"] = jsonParse;
        return obj; 
    }
```

`atob`/`btoa` modules shouldn't be used on the Node.js side (it's deprecated in favour of the `Buffer` class).

In fact, we should import the decoding functionality from msal-node & msal-common directly, or duplicate that code if its not exported. What we need here is the `extractTokenClaims` method from the [AuthToken](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/src/account/AuthToken.ts) class in msal-common, using the [base64decode](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/38b7fe045288993541250f240a8df081982d357e/lib/msal-node/src/crypto/CryptoProvider.ts#L49) implementation from msal-node.

Reply:
> `atob`/`btoa` modules shouldn't be used on the Node.js side (it's deprecated in favour of the `Buffer` class).
> 
> In fact, we should import the decoding functionality from msal-node & msal-common directly, or duplicate that code if its not exported. What we need here is the `extractTokenClaims` method from the [AuthToken](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/src/account/AuthToken.ts) class in msal-common, using the [base64decode](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/38b7fe045288993541250f240a8df081982d357e/lib/msal-node/src/crypto/CryptoProvider.ts#L49) implementation from msal-node.

Updated; Thanks for the feedback!